src/network/MissionControlProtocol.cpp

```diff
@@ -351,6 +371,25 @@ void MissionControlProtocol::jointPowerRepeatTask() {
 	}
 }
 
+void MissionControlProtocol::jointPosReportTask() {
+	dataclock::time_point pt = dataclock::now();
+
+	while (true) {
+		for (auto cur = robot::types::name_to_jointid.begin();
+			 cur != robot::types::name_to_jointid.end(); cur++) {
+			robot::types::DataPoint<int32_t> jpos = robot::getJointPos(cur->second);
+			if (jpos.isValid()) {
+				auto frozenStr = cur->first;
+				std::string str(frozenStr.data(), frozenStr.size());
```

Comment:
Can these names be more descriptive? Maybe `jointNameFrozenStr` and `jointNameStr`?